src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/V2FeedListResourceProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.LocalRepositories;
+
+namespace NuGet.Protocol.LegacyFeed
+{
+    public class V2FeedListResourceProvider : ResourceProvider
+    {
+
+        public V2FeedListResourceProvider() : base(
+            typeof (ListResource),
+            nameof(V2FeedListResourceProvider),
+            nameof(LocalPackageListResourceProvider))
+        {
+        }
+
+        public override async Task<Tuple<bool, INuGetResource>> TryCreate(SourceRepository source,
+            CancellationToken token)
+        {
+            ListResource resource = null;
+
+            if (await source.GetFeedType(token) == FeedType.HttpV2 ||
+                (source.PackageSource.IsHttp &&
+                 !source.PackageSource.Source.EndsWith(".json", StringComparison.OrdinalIgnoreCase)))
```

Comment:
Is this logic duplicated from somewhere else?
When would GetFeedType be "wrong" forcing us to use the second case here?